crates/eframe/src/epi.rs

```diff
@@ -196,6 +196,15 @@ pub trait App {
     fn persist_egui_memory(&self) -> bool {
         true
     }
+
+    /// A filter applied to raw input before [`Self::update`]
+    ///
+    /// This allows for the manipulation or filtering of input events before they are processed by egui.
+    ///
+    /// This can be used to exclude specific keyboard shortcuts, mouse events, etc.
+    ///
+    /// Additionally, it can be used to add custom keyboard or mouse events generated by a virtual keyboard.
+    fn filter_raw_input_inplace(&mut self, _ctx: &egui::Context, _raw_input: &mut egui::RawInput) {}
```

Comment:
Can we come up with a better name? It is more than just filtering 🤔

Reply:
How abour `on_raw_input` or `raw_input_hijack` ?

Reply:
on_raw_input()
simulate_input()
send_input()
fake_input()

Reply:
I like `raw_input_hook`

Reply:
OK, it has been changed to `raw_input_hook`.